pkg/cmd/pulumi/pulumi.go

```diff
@@ -133,8 +142,15 @@ func NewPulumiCmd() *cobra.Command {
 			// Before exiting, if there is a new version of the CLI available, print it out.
 			jsonFlag := cmd.Flag("json")
 			isJSON := jsonFlag != nil && jsonFlag.Value.String() == "true"
-			if checkVersionMsg := <-updateCheckResult; checkVersionMsg != nil && !isJSON {
-				cmdutil.Diag().Warningf(checkVersionMsg)
+
+			select {
+			case checkVersionMsg := <-updateCheckResult:
+				if checkVersionMsg != nil && !isJSON {
+					cmdutil.Diag().Warningf(checkVersionMsg)
+				}
+			default:
+				// If the channel doesn't have anything in it yet, just ignore it
+				// and continue shutting down.
```

Comment:
Not sure if you are still working on it, but I think these changes should now be reverted.

Reply:
Yes, that code needed to be updated. I had to review things myself, but I think this snippet has the behavior we want:

```go
checkVersionMsg, ok := <-updateCheckResult
 if ok && checkVersionMsg != nil && !isJSON {
    ...
}
```

- If the channel is closed, it will immediately return and `ok` will be `false`.
- If the channel is opened, then it will block until there is a value to be read from the channel.